docs/docsite/rst/dev_guide/developing_locally.rst

```diff
@@ -5,41 +5,47 @@
 Adding modules and plugins locally
 **********************************
 
-.. contents::
-   :local:
+The easiest, quickest, and the most popular way to extend Ansible is to use a module or a plugin. You can create them or copy existing ones for local use. You can store a local module or plugin on your Ansible control node and share it with your team or organization. You can also share a local plugin or module by embedding it in a role and publishing it on Ansible Galaxy. If you are using roles on Ansible Galaxy, then you are already using local modules and plugins without realizing it.
 
-The easiest, quickest, and most popular way to extend Ansible is to copy or write a module or a plugin for local use. You can store local modules and plugins on your Ansible control node for use within your team or organization. You can also share a local plugin or module by embedding it in a role and publishing it on Ansible Galaxy. If you've been using roles off Galaxy, you may have been using local modules and plugins without even realizing it. If you're using a local module or plugin that already exists, this page is all you need.
+If you are using an existing module or plugin but Ansible can't find it, this page is all you need. However, if you want to create a plugin or a module, go to :ref:`developing_plugins` and :ref:`developing_modules_general` topics and then return to this page to know how to add it locally. 
 
-Extending Ansible with local modules and plugins offers lots of shortcuts:
+Extending Ansible with local modules and plugins offers lots of shortcuts such as:
 
 * You can copy other people's modules and plugins.
 * If you're writing a new module, you can choose any programming language you like.
 * You don't have to clone the main Ansible repo.
 * You don't have to open a pull request.
 * You don't have to add tests (though we recommend that you do!).
 
-To save a local module or plugin so Ansible can find and use it, drop the module or plugin in the correct "magic" directory. For local modules, use the name of the file as the module name: for example, if the module file is ``~/.ansible/plugins/modules/local_users.py``, use ``local_users`` as the module name.
+To save a local module or plugin such that Ansible can find and use it, add the module or plugin in the appropriate directory (the directories are specified in later parts of this topic). 
+
+.. contents::
+   :local:
 
 .. _modules_vs_plugins:
 
-Modules and plugins: what's the difference?
+What are modules and plugins?
```

Comment:
I would keep the original one.

Reply:
Thank you for the review, @Akasurde. I updated the title in the commit instance 0880239.